.buildkite/gen-pipeline.sh

```diff
@@ -28,7 +28,6 @@ tests=( \
        test-cpu-openmpi-py3_6-tf2_0_0-keras2_2_4-torch1_2_0-mxnet1_5_0-pyspark2_4_0 \
        test-cpu-openmpi-py2_7-tfhead-kerashead-torchhead-mxnethead-pyspark2_4_0 \
        test-cpu-openmpi-py3_6-tfhead-kerashead-torchhead-mxnethead-pyspark2_4_0 \
-       test-cpu-mpich-py3_6-tf1_14_0-keras2_2_4-torch1_1_0-mxnet1_5_0-pyspark2_4_0 \
```

Comment:
Why was this test removed?

Reply:
@tgaddair 
Looks like horovod only support openmpi, do not support mpich (it cause test fail).
Also, env `OMPI_COMM_WORLD_RANK` is only supported in openmpi.

Reply:
Oh, Spectrum MPI was added yesterday https://github.com/horovod/horovod/pull/1441
I doubt whether this version MPI can support `OMPI_COMM_WORLD_RANK`.

Reply:
One option is `run_func` only support opemMPI, what do you think ? @tgaddair

Reply:
@WeichenXu123, unit tests using mpich are passing for HEAD.  If they're failing with your change, we should figure out why and fix them (some Horovod users use mpich).  

I don't see any reason why `run_func` should only support OpenMPI.  Using the function I provided above, it should work for OpenMPI, Spectrum, and mpich.  Please see my comment above regarding Spectrum support.

Reply:
Looks like you may need to rebase to get the change for Spectrum MPI.